### PR TITLE
Delegated Manager

### PR DESCRIPTION
**Pull request by Drypto13**

### CODE REVIEW

File: contracts/modules/LoanOpenings/LoanOpenings.sol [hunk 1]
```diff
@@ -355,7 +355,12 @@ contract LoanOpenings is State, LoanOpeningsEvents, VaultController, InterestHan
             sentValues,
             isTorqueLoan
         );
-
+        (uint128 flag) = abi.decode(loanDataBytes, (uint128));
+        if(flag & DELEGATE_FLAG != 0) {
+            (, bytes[] memory payloads) = abi.decode(loanDataBytes, (uint128, bytes[]));
+            (address delegateAddress) = abi.decode(payloads[1], (address));
```
Comment (RomanHiden) [inline contracts/modules/LoanOpenings/LoanOpenings.sol]: you can probably decode it right into function without declaring variable
I wonder if compiler optimizes for unnecessary variable. many times wanted to investigate. please check

Comment (RomanHiden) [reply contracts/modules/LoanOpenings/LoanOpenings.sol]: so if it optimizes it we leave for readability. if not we prefer efficiency

Comment (Drypto13) [reply contracts/modules/LoanOpenings/LoanOpenings.sol]: I am not sure if it removes unnecessary variables but will just decode it where needed versus storing in variables anyways.

**Review by RomanHiden — APPROVED** [review]
looks good

**Review by ghost — COMMENTED** [review]
looks good

**Review by ghost — COMMENTED** [review]
Looks good

**Pull request merged**